File: src/main/java/xyz/mlhmz/financemanager/entities/Category.java
package xyz.mlhmz.financemanager.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Category {
    @Id
    @UuidGenerator
    private UUID uuid;

    private String title;

    private String description;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "category")
    @Builder.Default
    private List<Transaction> transactions = new ArrayList<>();

    @ManyToOne(optional = false)
    private OAuthUser user;
}

File: frontend/src/categories/Category.ts
import { z } from "zod";

export interface Category {
  uuid?: string;
  title?: string;
  description?: string;
  createdAt?: string;
  updatedAt?: string;
}

export const MutateCategory = z.object({
  title: z.string(),
  description: z.string().optional(),
});

export type MutateCategory = z.infer<typeof MutateCategory>
